lib/unexpectedMitm.js

```diff
@@ -721,14 +721,18 @@ module.exports = {
                     afterBlockRegistered = true;
                 }
 
-                return executeMitm(expect, subject).then(function (recordedExchanges) {
+                return executeMitm(expect, subject).spread(function (value, recordedExchanges) {
                     if (injectIntoTest) {
                         var injectionCallsite = determineInjectionCallsite(stack);
                         if (injectionCallsite) {
                             recordPendingInjection(injectionCallsite, recordedExchanges);
                         }
                     }
-                    return recordedExchanges;
+                    if (expect.flags['with extra info']) {
+                        return [value, recordedExchanges];
+                    } else {
+                        return value;
+                    }
```

Comment:
This means 'with extra info' doesn't quite return the same order of stuff as for 'with http mocked out' - that's probably totally fine given it's a totally separate assertion :)